### PR TITLE
replace characters invalid for mysql

### PR DESCRIPTION
**Pull request by SimonHeimberg**

### CODE REVIEW

**Comment by SimonHeimberg** [comment]
next try to fix #145 and similar

Not properly tested since unit-tests do not run on my machine. (They report phpDocumentator\Reflection\DocBlock not found.) And maybe there are better variants. For running the tests, and for fixing the issue.

File: tests/PHPCensor/Helper/CommandExecutorTest.php [hunk 1]
```diff
@@ -85,23 +85,27 @@ public function testFindBinary_ReturnsNullWihQuietArgument()
...
     public function testReplaceIllegalCharacters()
     {
+        $execClass = new \ReflectionObject($this->testedExecutor);
+        $replaceMeth = $execClass->getMethod('replaceIllegalCharacters');
+        $replaceMeth->setAccessible(true);
+
         $this->assertEquals(
             \Normalizer::normalize("start � end"),
-            \Normalizer::normalize($this->testedExecutor->replaceIllegalCharacters(
+            \Normalizer::normalize($replaceMeth->invoke($this->testedExecutor,
```
[the remark this answers is not in the document]
Comment (SimonHeimberg) [reply tests/PHPCensor/Helper/CommandExecutorTest.php]: hm, this works locally with php 7.1. Any hints (from a human)?

**Comment by corpsee** [comment]
Thank you! I will review soon. You can see unit tests on [TravisCI](https://travis-ci.org/php-censor/php-censor/jobs/337779517).

File: tests/PHPCensor/Helper/CommandExecutorTest.php [hunk 2]
```diff
@@ -85,23 +85,27 @@ public function testFindBinary_ReturnsNullWihQuietArgument()
 
     public function testReplaceIllegalCharacters()
     {
+        $execClass = new \ReflectionObject($this->testedExecutor);
+        $replaceMeth = $execClass->getMethod('replaceIllegalCharacters');
+        $replaceMeth->setAccessible(true);
```
Comment (SimonHeimberg) [inline tests/PHPCensor/Helper/CommandExecutorTest.php]: Not necessary, the method is public. I obviously tested a wrong version ...

**Comment by SimonHeimberg** [comment]
will redo, since this two commits are wrong

**Comment by corpsee** [comment]
@SimonHeimberg First, thank you for your research and help. Are you finish your fix? Can I merge request?

**Comment by SimonHeimberg** [comment]
question: should we do the mysql specific replacement here (or only for mysql, in db backend)?

To be tested locally.

**Comment by corpsee** [comment]
@SimonHeimberg What mysql specific replacement do you mean? Can you give example or explanation?

**Comment by corpsee** [comment]
I merged request to the `master`. Feel free to comment and reopen.

**Comment by SimonHeimberg** [comment]
Local tests seem to pass. I was first unsure why some logs were still truncated. Something else was probably wrong.
Sorry for the delay.

mysql specific replacement: because mysql does not support 3 bytes wide unicode characters, see https://github.com/php-censor/php-censor/pull/149/files#diff-4f445c5eb1b18366c6f2c1b3b9194fffR182

**Comment by corpsee** [comment]
@SimonHeimberg Ok, thank you!

**Pull request merged**